### ISSUE TITLE
When Trident with loyality is in the weapon slot, it duplicates after throwing

### ISSUE THREAD

**Issue by Honza572** [TheRedBrain/rpg-inventory#16]
It would be nice to fix this for all items (so mods won't have problems)

**Comment by Honza572** [comment]
Possible solution that came to my mind: Make the mod create a file OUTSIDE of the .jar mod, preferably in config folder, that will remember IDs of all items you picked up. Then simply check the IDs only for the 4 usable slots that have this bug (the weapon/shield) and after use of an item from that slot, somehow check if the id of the entity-item matches the id in the inventory.
Oh and also ofc make it only for unstackable items, so if there will be something stackable it won't delete the whole stack.





Also the only thing I can do in Java is Hello World so maybe this idea is nonsense

**Comment by Honza572** [comment]
oh sorry I accidentally closed this I'm on mobile rn😅

**Comment by Honza572** [comment]
hey open

**Comment by TheRedBrain** [comment]
Apparently I forgot to respond, oops 😅 

I am looking into the issue, but I will most likely wait for 1.21.2. I expect that the TridentItem (in the code) will see some changes, which might make a solution easier.

For now I recommend "equipping" tridents in the hotbar.

**Comment by Honza572** [comment]
@TheRedBrain no worries, will there be a 1.21 fixed version too?

**Comment by TheRedBrain** [comment]
I don't know for sure yet

**Comment by TheRedBrain** [comment]
It's been a while, but I just checked if this is still an issue in the current dev version for 1.21.1.

I can no longer reproduce the duplication issue. The trident is still not returning to the main hand slot, I'm still looking into that.

**Comment by TheRedBrain** [comment]
Yeah returning a loyalty trident to the mainhand slot would require some mixin magic, which I sadly don't have time for right now. I will keep that feature on my todo list and I will hopefully be able to add it in newer versions, especially after the trident becomes component-driven. I'll close this issue, since the duplication glitch seems to be fixed.

If it should appear again, please let me know/re-open this issue :)
